pkg/reconciler/nscert/nscert_test.go

```diff
@@ -191,8 +193,12 @@ func TestReconcile(t *testing.T) {
 }
 
 func TestUpdateDomainTemplate(t *testing.T) {
-	defer ClearAll()
-	ctx, _, _, reconciler, watcher := newTestSetup(t)
+	ctx, cancel, _, _, reconciler, watcher := newTestSetup(t)
+	defer func() {
+		cancel()
+		time.Sleep(3 * time.Second)
```

Comment:
I wonder if we should make `SetupFakeContextWithCancel(t)` return a cancel function with this built in for now.

Reply:
I commented on this, since not all the tests suffer from this, making all of them delayed sounds a bit harsh.